src/v/cluster/rm_stm.cc

```diff
@@ -1004,11 +1004,9 @@ void rm_stm::load_snapshot(stm_snapshot_header hdr, iobuf&& tx_ss_buf) {
         _log_state.aborted.emplace(entry.pid, entry);
     }
     for (auto& entry : data.seqs) {
-        auto pid_seq = _log_state.seq_table.find(entry.pid);
-        if (pid_seq == _log_state.seq_table.end()) {
-            _log_state.seq_table.emplace(entry.pid, entry);
-        } else if (pid_seq->second.seq < entry.seq) {
-            pid_seq->second = entry;
+        auto [seq_it, _] = _log_state.seq_table.try_emplace(entry.pid, entry);
+        if (seq_it->second.seq < entry.seq) {
+            seq_it->second = entry;
```

Comment:
this hunk doesn't look like it is related to removing seq_stm. can be plucked out and squashed into the previous commit that adds the rm_stm?